RxCocoa/Common/DelegateProxyType.swift

```diff
@@ -210,6 +217,16 @@ extension DelegateProxyType {
             proxy.setForwardToDelegate(nil, retainDelegate: retainDelegate)
         }
     }
+    
+    public static func extendProxy<Object: AnyObject>(with factory: @escaping ((Object) -> AnyObject)) {
+        MainScheduler.ensureExecutingOnScheduler()
+        _ = delegateProxyFactory.extended(factory: factory)
+    }
+    
+    public static func createProxyForObject(_ object: AnyObject) -> AnyObject {
+        MainScheduler.ensureExecutingOnScheduler()
```

Comment:
It seems to me that this is in wrong place. Somebody could made a direct call to `delegateProxyFactory` and avoid this check.